parse-debian-clang.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -x
+
+parse_parameters() {
+    while [[ ${#} -gt 0 ]]; do
+        case ${1} in
+            -c | --check) action=check ;;
+        esac
+        shift
+    done
+
+    # Default action is check
+    [[ -z ${action} ]] && action=check
+}
+
+parse_clang_version() {
+    # The format of clang --version with apt.llvm.org builds looks like:
+    # $ clang-9 --version
+    # clang version 9.0.0-svn356030-1~exp1+0~20190313082415.405~1.gbp505100 (trunk)
+    # Target: x86_64-pc-linux-gnu
+    # Thread model: posix
+    # InstalledDir: /usr/bin
+    clang_date=$(clang --version | head -n1 | cut -d '~' -f 3 | cut -d . -f 1)
+
+    # Next, we need to parse the date into a format the date binary can understand
+    # We use bash substring expansion: https://wiki.bash-hackers.org/syntax/pe#substring_expansion
+    clang_date="${clang_date:0:4}-${clang_date:4:2}-${clang_date:6:2} ${clang_date:8:2}:${clang_date:10:2}:${clang_date:12:2}"
```

Comment:
I worry this is going to be fragile, but since it should fail loudly (i.e. clang_date will never be blank), I think it's okay.